cpp/include/cugraph/edge_partition_edge_property_device_view.cuh

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cugraph/edge_src_dst_property.hpp>
+#include <cugraph/utilities/device_functors.cuh>
+
+#include <raft/span.hpp>
+
+#include <thrust/binary_search.h>
+#include <thrust/distance.h>
+#include <thrust/execution_policy.h>
+#include <thrust/fill.h>
+#include <thrust/iterator/iterator_traits.h>
+#include <thrust/optional.h>
+
+namespace cugraph {
+
+namespace detail {
+
+template <typename edge_t, typename ValueIterator>
+class edge_partition_edge_property_device_view_t {
```

Comment:
Would edge_property_partition_device_view_t read better?

Reply:
So, there is `edge_partition_endpoint_property_device_view_t` as well.

This name implies a device view object for edge properties in an edge partition (and `edge_partition_endpoint_property_device_view_t` means a device view object for edge endpoint (source/destination) properties in an edge partition).

Reply:
And there will be `vertex_partition_vertex_property_device_view_t` in the future as well (to support vertex masking)... These names should be considered in this context.

`edge_partition` or `vertex_partition` comes first to emphasize that these work on a vertex partition or an edge partition.

"edge_property_partition" or "edge_endpoint_partition" somewhat sounds like we are partitioning edge properties or edge endpoints instead of edges... Edge partitioning proceeds and edge properties or endpoint properties just follow this edge partition.

Based on this assumption, let me know if you have suggestions for better names.

Reply:
According the what you describe, it probably best to adhere with this naming convention.